### PR TITLE
Made some changes

### PR DESCRIPTION
**Pull request by dmnjohns**
Ch-ch-ch-ch-ch-check it out!

Changes to #1241

### CODE REVIEW

File: repose-aggregator/components/filters/herp/src/main/scala/org/openrepose/filters/herp/HerpFilter.scala [hunk 1]
```diff
@@ -112,55 +81,50 @@ class HerpFilter @Inject()(configurationService: ConfigurationService) extends F
...
-              if (keySplit.size > 1) {
-                x.getOrElse(keySplit(1), Array("")).filter { s => pattern.findFirstIn(s).isDefined}.nonEmpty
-              } else {
-                false
-              }
+    def filter(valuesMap: Map[String, Any]): Boolean = {
+      filtersOut.exists { fo =>
+        fo.getMatch.asScala.forall { m =>
+          val field = m.getField
+          val pattern = m.getRegex.r.pattern
```
Comment (dmnjohns) [inline repose-aggregator/components/filters/herp/src/main/scala/org/openrepose/filters/herp/HerpFilter.scala]: This should not happen on every request. Do it once when we load the configuration.

File: repose-aggregator/components/filters/herp/src/main/scala/org/openrepose/filters/herp/HerpFilter.scala [hunk 2]
```diff
@@ -112,55 +81,50 @@ class HerpFilter @Inject()(configurationService: ConfigurationService) extends F
...
+      "requestQueryString" -> httpServletRequest.getQueryString,
+      "parameters" -> nullIfEmpty(translateParameters()),
+      "timestamp" -> System.currentTimeMillis,
+      "responseCode" -> httpServletResponse.getStatus,
+      "responseMessage" -> Try(HttpStatus.valueOf(httpServletResponse.getStatus).name).getOrElse("UNKNOWN"),
+      "guid" -> java.util.UUID.randomUUID.toString,
+      "serviceCode" -> serviceCode,
+      "region" -> region,
+      "dataCenter" -> dataCenter
     )
```
Comment (dmnjohns) [inline repose-aggregator/components/filters/herp/src/main/scala/org/openrepose/filters/herp/HerpFilter.scala]: This map should convert all Scala types to Java types.

Comment (dmnjohns) [inline repose-aggregator/components/filters/herp/src/main/scala/org/openrepose/filters/herp/HerpFilter.scala]: This should be done now.

File: repose-aggregator/components/filters/herp/src/main/scala/org/openrepose/filters/herp/HerpFilter.scala [hunk 3]
```diff
@@ -112,55 +81,50 @@ class HerpFilter @Inject()(configurationService: ConfigurationService) extends F
...
+
+          valuesMap.get(field) match {
+            case Some(value: Int) => pattern.matcher(value.toString).matches()
+            case Some(value: Long) => pattern.matcher(value.toString).matches()
+            case Some(value: String) => pattern.matcher(value).matches()
+            case Some(value: Array[String]) => value.exists(pattern.matcher(_).matches())
+            case Some(value: Map[String, Array[String]]) => ???
+            case Some(_) => false // we don't know how to match against the provided type
+            case None => false
           }
```
Comment (dmnjohns) [inline repose-aggregator/components/filters/herp/src/main/scala/org/openrepose/filters/herp/HerpFilter.scala]: We either need a case for Java types, or need to convert to Scala types for this match.

**Comment by adrianjgeorge** [comment]
Is this redundant with the main branch now?

**Comment by dmnjohns** [comment]
Essentially. The behavior is replicated, at least.

**Pull request closed without merging**